### PR TITLE
add managed identity support to azure storage queue

### PR DESCRIPTION
**Pull request by jasonwbarnett**
The title is self-explanatory.

I added another commit because flake8 tests were failing.

### CODE REVIEW

File: requirements/extras/azurestoragequeues.txt [hunk 1]
```diff
@@ -1 +1,2 @@
 azure-storage-queue>=12.2.0
+azure-identity>=1.5.0
```
Comment (auvipy) [inline requirements/extras/azurestoragequeues.txt:2]:
```suggestion
azure-identity>=1.12.0
```

**Review by auvipy — CHANGES_REQUESTED** [review]
why 1.5 and not 1.12.0

**Comment by jasonwbarnett** [comment]
> why 1.5 and not 1.12.0

I just thought to give a broader support for folks who might be running an older version. In reality I don't know what this is used for so it is fine to be the latest as well. I'll update the PR later when I'm done traveling.

**Comment by jasonwbarnett** [comment]
@auvipy I forgot to mention that this has been tested and is working in our production environment.

**Comment by jasonwbarnett** [comment]
I reviewed the failing integration tests and it appears to be only redis and completely unrelated to this PR. Is that true?

**Comment by auvipy** [comment]
yes that is true

**Comment by auvipy** [comment]
should we also document the change?

**Comment by jasonwbarnett** [comment]
> yes that is true

👍 

> should we also document the change?

Yes. I thought I already did in the header of the file. Is there somewhere else to document the change? Please let me know and I will write the docs.

**Comment by jasonwbarnett** [comment]
I tried refining the documentation even further. Let me know if I formatted them correctly, I've never used rst.

**Comment by auvipy** [comment]
here https://github.com/celery/kombu/pull/1634 mostly pypy integration tests are failing

**Comment by jasonwbarnett** [comment]
> here #1634 mostly pypy integration tests are failing

@auvipy Should I do anything more on this PR or is this PR acceptable in it's current state to be eventually merged when you are ready from a project perspective?

**Comment by auvipy** [comment]
let me stabilize the CI little bit more, then you can pull from main branch

**Review by thedrow — APPROVED** [review]
LGTM

**Review by auvipy — APPROVED** [review]

**Pull request merged**